airflow/jobs/scheduler_job.py

```diff
@@ -1633,6 +1634,9 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
                 )
             )
         )
+        orphaned_datasets_count = 0
         for dataset in orphaned_dataset_query:
             self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
             dataset.is_orphaned = expression.true()
+            orphaned_datasets_count += 1
+        Stats.incr("dataset.orphaned", count=orphaned_datasets_count)
```

Comment:
I think 
```suggestion
        for dataset in orphaned_dataset_query:
            self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
            dataset.is_orphaned = expression.true()
        Stats.gauge("dataset.orphaned", len(orphaned_dataset_query))
```

This should be a gauge instead of a counter, because datasets can also be "un-orphaned".